Use IPython 3.0

bollwyvl opened the pull request:
Hooray IPython 3.0 :cake:! Welcome to the future :monorail:!

This upgrades nbviewer's IPython requirement to `IPython[notebook]>=3.0`.

Since all good things come in threes, I'd love to see (IPython, Bootstrap & Reveal (ahem, #390)) 3.x all wrapped up as a 0.3.0 release!

bollwyvl commented:
Error appears to be Travis->Github DNS...

bollwyvl commented:
Hooray, build fixed. Docker build improved somewhat as well.... mainly to prevent re-pip/bower-ing after every file change for local test... though this could be taken out, if that's too far outside the scope.

bollwyvl commented:
Gah, or the build is not fixed. Again, seeing lots fo `40x` from github. My oh my.

rgbkrk commented on Dockerfile:
```diff
@@ -28,23 +28,27 @@ RUN npm install -g bower less
 RUN pip install -r requirements.txt
 
+ADD ./tasks.py /srv/nbviewer/
+ADD ["/nbviewer/static/bower.json", "/nbviewer/static/.bowerrc", "/srv/nbviewer/nbviewer/static/"]
```
Well this is new to me. How does `ADD` work with a list?

rgbkrk replied:
Ah, it's intended to be `["<src>"... "<dest>"]`. Should these be `./nbviewer/static/...`?

bollwyvl replied:
You're right! Will fix!

rgbkrk commented on README.md:
````diff
@@ -63,6 +63,18 @@ docker build -t nbviewer .
+
+```shell
+docker-compose up
+```
````
Thank you for doing this! I've been waiting on compose to release, had not realize it had shipped.

bollwyvl replied:
Yeah they were very sneaky. It was confusing for a while as they still were using the fig gh.

rgbkrk commented on requirements.txt:
```diff
@@ -1,4 +1,3 @@
-tornado>=3.1.1
```
Technically this makes sure our underlying server matches tornado. I couldn't imagine IPython notebook changing (like it did from twisted) again, but still.

bollwyvl replied:
True. Will fix!

rgbkrk commented:
Looking good. If the docker build is going well, I say ship this. The only comment I made that I care about a response is the `ADD` with a list.

bollwyvl commented:
Addressed above comments. Changed the links to bower stuff... was definitely a typo, though ADD will [only, ahem, add](http://stackoverflow.com/questions/27068596/how-to-include-files-outside-of-dockers-build-context) things inside the current build context/dir. But it looked weird, for sure, and the `./` is easier to understand!

Eyeing the shiny green button, but wouldn't mind some more :eyes:! In the meantime will start working up the bootstrap3 stuff!

rgbkrk commented:
Looks great to me. Thanks @bollwyvl!
